use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+  Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+  Pipeline pipeline = Pipeline.create(options);
+
+  final Instant stopMinTimestamp = new Instant(minFmt.parseMillis(options.getStopMin()));
+  final Instant startMinTimestamp = new Instant(minFmt.parseMillis(options.getStartMin()));
+
+  // Read 'gaming' events from a text file.
+  pipeline.apply(TextIO.Read.from(options.getInput()))
+    // Parse the incoming data.
+    .apply("ParseGameEvent", ParDo.of(new ParseEventFn()))
+
+    // Filter out data before and after the given times so that it is not included
+    // in the calculations. As we collect data in batches (say, by day), the batch for the day
+    // that we want to analyze could potentially include some late-arriving data from the previous
+    // day. If so, we want to weed it out. Similarly, if we include data from the following day
+    // (to scoop up late-arriving events from the day we're analyzing), we need to weed out events
+    // that fall after the time period we want to analyze.
+    // [START DocInclude_HTSFilters]
+    .apply("FilterStartTime", Filter.by(
+        (GameActionInfo gInfo)
+            -> gInfo.getTimestamp() > startMinTimestamp.getMillis()))
+    .apply("FilterEndTime", Filter.by(
+        (GameActionInfo gInfo)
+            -> gInfo.getTimestamp() < stopMinTimestamp.getMillis()))
+    // [END DocInclude_HTSFilters]
+
+    // [START DocInclude_HTSAddTsAndWindow]
+    // Add an element timestamp based on the event log, and apply fixed windowing.
+    .apply("AddEventTimestamps",
+           WithTimestamps.of((GameActionInfo i) -> new Instant(i.getTimestamp())))
+    .apply("FixedWindowsTeam", Window.<GameActionInfo>into(
+        FixedWindows.of(Duration.standardMinutes(options.getWindowDuration()))))
+    // [END DocInclude_HTSAddTsAndWindow]
+
+    // Extract and sum teamname/score pairs from the event data.
+    .apply("ExtractTeamScore", new ExtractAndSumScore("team"))
+    .apply("WriteTeamScoreSums",
+      new WriteWindowedToBigQuery<KV<String, Integer>>(options.getTableName(),
+          configureWindowedTableWrite()));
+
+  pipeline.run().waitUntilFinish();
+}
+```
+
+### Limitations
+
+As written, `HourlyTeamScore` still has a limitation:
+
+* `HourlyTeamScore` still has **high latency** between when data events occur (the event time) and when results are generated (the processing time), because, as a batch pipeline, it needs to wait to begin processing until all data events are present.
+
+
+## LeaderBoard: Streaming Processing with Real-Time Game Data
+
+One way we can help address the latency issue present in the `UserScore` and `HourlyTeamScore` pipelines is by reading the score data from a streaming source. The `LeaderBoard` pipeline introduces streaming processing by reading the game score data from [Google Cloud Pub/Sub](https://cloud.google.com/pubsub/), rather than from a file on the game server.
```

Comment:
"an unbounded source that produces an infinite amount of data" instead of pubsub